job/runner.go

```diff
@@ -0,0 +1,131 @@
+package job
+
+import (
+	"errors"
+	"os/exec"
+	"time"
+)
+
+type JobRunner struct {
```

Comment:
@samdolan So I split this out, which helps with what I was talking to you about on Friday.